bgp/bgp.proto

```diff
@@ -40,6 +40,13 @@ message ClearBGPNeighborRequest {
     SOFT = 0;         // Send route-refresh and reapply policy.
     SOFTIN = 1;       // Re-apply inbound policy on stored Adj-RIB-In.
     HARD = 2;         // Teardown and restart TCP connection.
+    HARD_RESET = 3;         // Teardown, restart TCP connection, Send Cease 
+    // notification (code 6) with subcode 4. However, if the "N" bit for 
+    // Graceful-Restart" is set, then encapsulate the original Notification
+    // message in the data portion of the Subcode 9 message as per 
+    // rfc8538#section-3.1 and also Flush the routes.
+    GRACEFUL_RESET = 4;     // Teardown, restart TCP connection, Send Cease 
+    // notification (code 6) with subcode 4.
```

Comment:
Formatting and ordering change for the description

```suggestion
    // If the "N" bit for Graceful-Restart" is set, then encapsulate the original 
    // Notification message in the data portion of the Subcode 9 message per 
    // rfc8538#section-3.1 and Flush routes associated with this neighbor.
    // Otherwise, send BGP Cease notification (code 6) with subcode 4 then 
    // teardown and restart the TCP connection.
    HARD_RESET = 3;
    // Send BGP Cease notification (code 6) with subcode 4 then 
    // teardown and restart the TCP connection.
    GRACEFUL_RESET = 4;     
```

Reply:
It seems we want two things:

1. Hard reset the connection and override graceful restart, if it is enabled
2. Hard reset the TCP connection, but maintain graceful restart, if it is enabled.

The current field (`HARD = 2`) is ambiguous in the presence of graceful restart.  I suggest we deprecate it.  How does that sound?

How about we name the two new fields like this:

`HARD_RESET_GRACEFUL` = 3    // Reset and restart the TCP connection, preserving routes if  graceful restart is enabled
`HARD_RESET_GRACEFUL_OVERRIDE` = 3   // Reset and restart the TCP connection, overriding graceful restart per RFC8538 Section 3.1


This avoids restating or interpreting the relevant RFC (which I agree is a little difficult to read, so I see why it's tempting to clarify it here)